Sources/Apollo/NormalizedCache.swift

```diff
@@ -1,5 +1,46 @@
 import Foundation
 
+/// An object that drives behavior of clear operations on a given cache.
+public struct CacheClearingPolicy: Equatable {
+  /// Clears all records in the cache.
+  public static let allRecords = CacheClearingPolicy(.allRecords)
+  /// Clears all records whose key matches the provided glob pattern.
+  /// - Parameter pattern: The glob pattern to use for key matching. For example `*pollo` will match both `Apollo` and `pollo`.
+  public static func allMatchingKeyPattern(_ pattern: String) -> CacheClearingPolicy { .init(.allMatchingKeyPattern(pattern)) }
+  /// Clears the the first (oldest) records in the cache up to the given limit.
+  /// - Parameter k: The number of records to remove from the cache.
```

Comment:
i would strongly recommend making this something like `numberOfRecords` instead of `k`, particularly since you're using the `_` for the external parameter.